Docs/Archiving.md

```diff
@@ -0,0 +1,68 @@
+# Archiving Notes
+
+There is a preliminary implementation of NSKeyedArchiver and NSKeyedUnarchiver which should be compatible with the OS X version.
+
+* The implementation of NSStringFromClass() and NSClassFromString() currently uses private Swift API and does not support encoding classes that are not exported (i.e. cannot be looked up with dlsym()
+
+* NSKeyedUnarchiver reads the entire plist into memory before constructing the object graph, it should construct it incrementally as does Foundation on OS X
+
+* NSConcreteValue is only partially implemented, it does not have encoding complex types such as NSRange, NSPoint, NSRect, etc
```

Comment:
darwin special cases these types (primarily for speed) which I bet we can do the same easily